compair/tests/api/test_lti_launch.py

```diff
@@ -912,7 +912,108 @@ def test_lti_status(self):
             self.assertTrue(status['assignment']['exists'])
             self.assertEqual(status['assignment']['id'], assignment.uuid)
 
-    def test_cas_auth_via_lti_launch(self):
+    def test_new_cas_auth_via_lti_launch(self):
+        auth_data = ThirdPartyAuthTestData()
+
+        lti_consumer = self.lti_data.get_consumer()
+        roles = {
+            "Instructor": (SystemRole.instructor, CourseRole.instructor),
+            "urn:lti:role:ims/lis/Instructor": (SystemRole.instructor, CourseRole.instructor),
+            "Faculty": (SystemRole.instructor, CourseRole.instructor),
+            "urn:lti:role:ims/lis/Faculty": (SystemRole.instructor, CourseRole.instructor),
+            "Staff": (SystemRole.instructor, CourseRole.instructor),
+            "urn:lti:role:ims/lis/Staff": (SystemRole.instructor, CourseRole.instructor),
+            "Student": (SystemRole.student, CourseRole.student),
+            "urn:lti:role:ims/lis/Student": (SystemRole.student, CourseRole.student),
```

Comment:
Is there a list to verify the roles?  From [this guide](http://www.imsglobal.org/specs/ltiv1p0/implementation-guide#_Toc261271984), we have
`urn:lti:instrole:ims/lis/Student` and `urn:lti:role:ims/lis/Learner` but no `urn:lti:role:ims/lis/Student`.

Similar case to `urn:lti:role:ims/lis/Faculty` and `urn:lti:role:ims/lis/Staff`

Reply:
The list from the guide is accurate as far as roles we might care about (I believe there is a longer list somewhere with even more roles we don't need to care about). What the Tool Consumers actually send in lti launchs and membership requests is somewhat inconsistent (sometime they send the simple shortened role ex `Student` or the longer role `urn:lti:instrole:ims/lis/Student`. they also send a single value or array). 

In order to avoid potential issues with this and keep it as simple as possible, our implementation actually only tries to look for higher roles (lowercase `faculty`, `staff`, `instructor`, and `teachingassistant` in any part of the role string). Otherwise we automatically assume the user is a student (and system/course permissions can be fixed the instructors/admins as needed). This is why I don't have a test for leaner, content developer, etc since they all fall into the student umbrella by default